OpenRA.Game/Graphics/Util.cs

```diff
@@ -191,6 +223,69 @@ public static void FastCopyIntoSprite(Sprite dest, Png src)
 			}
 		}
 
+		/// <summary>Rotates a quad (rectangle) rotated along its Z-axis</summary>
+		/// <param name="tl">The top left vertex of the quad</param>
+		/// <param name="size">A float3 containing the X, Y, and Z lengths of the quad</param>
+		/// <param name="rotation">The number of radians to rotate by</param>
+		/// <returns>An array of four vertices representing the rotated quad (top-left, top-right, bottom-right, bottom-left)</returns>
+		public static float3[] RotateQuad(float3 tl, float3 size, float rotation)
```

Comment:
IMO its unintuitive for this method to take the top-left corner and size, but then rotate about its center (and in fact I had to delete earlier review comments where I pointed out what I thought were bugs before I read this). It would be more natural to take either the center and its size, or the top-left and bottom-right corners.

Reply:
I see what you mean, however using the top-left corner fits nicely with the `SpriteRenderable` call, where top-left corner is easily available. Also while `SpriteRenderable` does not pass in Size directly (it passes in bottom-right minus top-left), `UISpriteRenderable` and the `BoundingRectangle()` function both pass in `sprite.Size` as the second parameter, so changing this to the bottom-right corner would add further calculations. So while it might be a little unintuitive, I think the alternatives would require more calculations than at present.

Reply:
That doesn’t make sense… the first this this method does is to calculate the center, since it can’t work with the top-left directly.

Reply:
> That doesn’t make sense… the first this this method does is to calculate the center, since it can’t work with the top-left directly.

Not more calculations performance wise, but it would mean the center calculation would exist in more places than it currently does, since each function call would need to calculate it beforehand. I think this is messier and more prone to error, but if you really want it I guess it can be done. Though given this doesn't change the functionality at all I wonder whether this might be overcomplicating what is a fairly simple function.

Reply:
Ok. Can we please instead change the summary to something like `/// <summary>Rotates a quad about its center in the x-y plane.</summary>

Reply:
Sure, thanks for this. Done.